packages/desmos-std/src/types.rs

```diff
@@ -1,13 +1,30 @@
-use crate::query_types::DesmosQueryWrapper;
+use crate::query_router::DesmosQueryRouter;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-pub type Deps<'a> = cosmwasm_std::Deps<'a, DesmosQueryWrapper>;
-pub type DepsMut<'a> = cosmwasm_std::DepsMut<'a, DesmosQueryWrapper>;
+pub type Deps<'a> = cosmwasm_std::Deps<'a, DesmosQueryRouter>;
+pub type DepsMut<'a> = cosmwasm_std::DepsMut<'a, DesmosQueryRouter>;
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum DesmosRoute {
     Subspaces,
     Profiles,
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct PageRequest {
+    key: Vec<u8>,
+    offset: u64,
+    limit: u64,
+    count_total: bool,
+    reverse: bool,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct PageResponse {
+    next_key: Option<String>,
+    total: String,
```

Comment:
```suggestion
     #[serde(default)]
    total: u64,
```
The total field type is u64. In addition, the total field would be skipped when total is equal to 0 so #[serde(default)] tag is needed.

Reply:
I tried to use `u64` as well, but it appears that the `PageResponse` JSON is Serialized as follow:
```json
"pagination": {
    "next_key": null,
    "total": "4"
  }
```
So the total is basically a `String`. This will make the deserialisation to u64 fail. Also when it's 0, it will be serialised as:
```json
"pagination": {
    "next_key": null,
    "total": "0"
  }
```

I've tested it locally and it happens to not work at all with `u64` unfortunately

Reply:
You are right, I used default json marshal. Using cdc marshaling will get right result.

Reply:
@braze
Finally I found a better solution which is using cosmwasm_std::Uint64 and cosmwasm_std::Binary.
The pagination could be:
```rust
pub struct PageResponse {
    next_key: Option<Binary>,
    total: Uint64,
}
```